### PR TITLE
SEC-8 - Locking down unsecured and unused resources.

### PR DESCRIPTION
**Pull request by awilsonCurt**

### CODE REVIEW

File: index.go [hunk 1]
```diff
@@ -295,39 +300,42 @@ func main() {
 		r.Get("/search/geo/:latitude/:longitude", dealers_ctlr.SearchLocationsByLatLng)
 	})
 
+	//Creating, updating, and deleting FAQs are done in GoAdmin directly
```
Comment (WileESpaghetti) [inline index.go:305]: If that's the case, these routes seem like they may be good candidates for deprecating. I've been creating tickets similar to API-105 for endpoints to keep track of routes that might not be used anymore. Sometime down the road when we get enough enpoints (and free time ;) ) we were going to evaluate further and see if we can actually deprecate/remove them

File: index.go [hunk 2]
```diff
@@ -372,106 +381,113 @@ func main() {
 		r.Get("", part_ctlr.All)
 	})
 
+	//Creating, updating, and Deleting of salesRep entities is all done in GoAdmin directly
 	m.Group("/salesrep", func(r martini.Router) {
```
Comment (WileESpaghetti) [inline index.go]: Here's more routes that we might could have deprecation tickets made in jira

File: index.go (hunk 1 above)
Comment (WileESpaghetti) [reply index.go:305]: Also applies to
L80
L88
L119
L314
L347
L384
L441
L523

Comment (awilsonCurt) [reply index.go:305]: Yes, we can most likely remove large chunks of this API with little to no effect

File: index.go [hunk 3]
```diff
@@ -226,9 +231,9 @@ func main() {
 		r.Get("/user/:id", customer_ctlr.GetUserById)
 		r.Post("/user/:id", customer_ctlr.UpdateCustomerUser)
```
Comment (dgallegos) [inline index.go]: Do customers use their own key to update this?

File: index.go [hunk 4]
```diff
@@ -195,22 +196,26 @@ func main() {
...
 	m.Group("/cache", func(r martini.Router) { // different endpoint because partial matching matches this to another excused route
 		r.Get("/key", cache.GetByKey)
 		r.Get("/keys", cache.GetKeys)
-		r.Delete("/keys", cache.DeleteKey)
+		r.Delete("/keys", middleware.InternalKeyAuthentication, cache.DeleteKey)
 	})
 
+	//No lockdown of customer related endpoints for now
 	m.Group("/cust", func(r martini.Router) { // different endpoint because partial matching matches this to another excused route
 		r.Post("/user/changePassword", customer_ctlr.ChangePassword)
```
Comment (dgallegos) [inline index.go:208]: Do customers use their own keys to change their passwords?

File: index.go (hunk 3 above)
Comment (awilsonCurt) [reply index.go]: I'll check in the logs, but almost everything seems to just use one key, only difference is that the dealer site seems to use the customers key when doing requests.

File: index.go (hunk 4 above)
Comment (awilsonCurt) [reply index.go:208]: From the logs it looks like almost all instances of this are 500s, and none of them have keys at all. So not sure.

File: index.go (hunk 3 above)
Comment (awilsonCurt) [reply index.go]: For this, I can't find any instance of it in the logs, but looking at the code it seems like they might be intended to use their own key.

Comment (dgallegos) [reply index.go]: From the code, can you update another user's information with another key?

Comment (dgallegos) [reply index.go]: Try to see if you can update my information with your key.

File: index.go (hunk 4 above)
Comment (dgallegos) [reply index.go:208]: Try to see if you can update my information with your key.

**Review by dgallegos — APPROVED** [review]

**Pull request merged**